### ISSUE TITLE
Pressure Advance or Linear Advance Data Field

### ISSUE THREAD

**Issue by Ktmdude88** [Donkie/Spoolman#20]
Please include a data field for PA ([Pressure Advance for Klipper users](https://www.klipper3d.org/Pressure_Advance.html)) or LA ([Linear Advance for Marlin users](https://marlinfw.org/docs/features/lin_advance.html)).
Typical setting expected is 0 to 1.000 for PA, and typically 0 to 2.0 for LA.

This setting is spool specific and goes hand in hand with flow % for an individual spool.

Believe this was first mentioned by @locki-cz in issue #10.

**Comment by fbeauKmi** [comment]
Hi, first, I do appreciate your project. IMO , PA, flow, and other print parameters are not only spool specific but also printer/toolhead/nozzle diameter dependant. It would be nice to store this datas according to a toolhead reference. Till now I use a Klipper macro to store filament specific values in the printer  https://github.com/fbeauKmi/V2.3934/blob/main/klipper_config/macros/helpers/db_filament.cfg and call them in print_start. I planned to do to moonraker plugin but it's a bit out of my skills.

**Comment by Donkie** [comment]
Perhaps a new item type is in order then? Like a configuration set. You could then add new configs to specific combinations of vendor, filament, printer, toolhead, nozzle. 

Having a system like this instead of having parameters like "override extruder temperature" on the filament object would make it a lot more flexible.

I created a quick mockup, would this make sense?

![image](https://github.com/Donkie/Spoolman/assets/2332094/11ef648a-f9a7-4ade-8cfb-12268b278d83)

**Comment by fbeauKmi** [comment]
That looks really nice !. I am not sure to understand exactly how it works. 
But IMO, Pressure advance and extrude factor (Flow rate), for example, are spool related. I experienced filament : same brand/vendors/color with different values for PA and EM.
Does this criterias would be accessible or could be modified thru moonraker ? I use much more parameters but I have to admit it is overkill for many users. Here the list currently running on a printer : eg :
 Settings for ASA_AZUREFILM:
// _ bed_temp : 105.0
// _ chamber_min_temp : 30 (default)
// _ chamber_temp : 60 (default)
// _ extrude_factor : 0.95653
// _ extruder_temp : 250.0
// _ fan_speed : 50.0
// _ filament_sensor : 1.0 (default)
// _ filter_speed : 0.5 (default)
// _ max_flow : 26.0
// _ pa_smooth_time : 0.025
// _ pressure_advance : 0.035
// _ retract_length : 0.4
// _ retract_speed : 35.0
// _ soak_delay : 30.0 (default)
// _ unretract_extra_length : 0.0 (default)
// _ unretract_speed : 35.0
// _ z_adjust : 0.0 (default)

BTW, let me know if I can help !

**Comment by Donkie** [comment]
I realized that a lot of these settings must be tuned in the slicer, and overriding gcodes will be too late. So perhaps a slicer integration would be necessary then as well, which sounds like a daunting task given how many different slicers there are etc.

**Comment by Ktmdude88** [comment]
> I realized that a lot of these settings must be tuned in the slicer, and overriding gcodes will be too late. So perhaps a slicer integration would be necessary then as well, which sounds like a daunting task given how many different slicers there are etc.

Correct, really the only overrides in my personal workflow are PA and the occasional extruder or bed temperature overrides after a print has been started but I forgot to change them.
